sklearn/utils/_plot.py

```diff
@@ -0,0 +1,53 @@
+from . import check_consistent_length, check_matplotlib_support
+from .multiclass import type_of_target
+from .validation import _check_pos_label_consistency
+from ._response import _get_response_values_binary
+
+
+class _BinaryClassifierCurveDisplayMixin:
+    """Mixin class to be used in Displays requiring a binary classifier.
+
+    The aim of this class is to make some validations regarding the estimator and the
+    target and gather the response of the estimator.
+    """
+
+    def plot(self, *, name=None):
+        check_matplotlib_support(f"{self.__class__.__name__}.plot")
+        name = self.estimator_name if name is None else name
+        return name
+
+    @classmethod
+    def from_estimator(
+        cls, estimator, X, y, *, response_method="auto", pos_label=None, name=None
+    ):
```

Comment:
Same remark for the ``from_estimator`` and ``from_predictions`` method. It's confusing to have methods with a same name that do different things.